### PR TITLE
Refactor `tests/test_mlflow_lazily_imports_ml_packages.py`

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

Refactor `tests/test_mlflow_lazily_imports_ml_packages.py` and remove the `--lazy-import` flag. This changes allows us to simplify `dev/run-python-flavor-tests.sh` and make it easier to check the test results.

## How is this patch tested?

Existing tests

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -228,9 +228,6 @@ jobs:
         env:
           SPARK_LOCAL_IP: 127.0.0.1
         run: |
-          # Install libopenblas-dev for mxnet 1.8.0.post0
-          sudo apt-get update -y
-          sudo apt-get install libopenblas-dev -y
```
Comment (harupy) [inline .github/workflows/master.yml:233]: We no longer run tests for `mxnet` in `run-python-flavor-tests.sh`.

File: pytest.ini [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [pytest]
-addopts = --cov=./mlflow --cov-report html --color=yes --durations=5 --showlocals --verbose
+addopts = --cov=./mlflow --cov-report html --color=yes --durations=10 --showlocals --verbose
```
Comment (harupy) [inline pytest.ini:2]: Increased `durations` to show more test results.

File: tests/test_mlflow_lazily_imports_ml_packages.py [hunk 3]
```diff
@@ -1,23 +1,14 @@
...
 
-def _get_loaded_packages():
-    return {k.split(".")[0] for k in sys.modules.keys()}
-
-
-# Run this test only when the `--lazy-import` flag is specified
-@pytest.mark.lazy_import
-def test_mlflow_lazily_import_ml_packages():
-    ml_packages = {
+@pytest.mark.parametrize(
```
Comment (harupy) [inline tests/test_mlflow_lazily_imports_ml_packages.py:6]: Parametrized this test to make it easier to debug.

File: dev/run-python-flavor-tests.sh [hunk 4]
```diff
@@ -1,23 +1,15 @@
...
-test $err = 0
+pytest \
+  tests/azureml \
+  tests/utils/test_model_utils.py \
+  tests/h2o \
+  tests/shap \
+  tests/paddle \
+  tests/tracking/fluent/test_fluent_autolog.py \
+  tests/autologging \
+  tests/test_mlflow_lazily_imports_ml_packages.py
```
Comment (harupy) [inline dev/run-python-flavor-tests.sh:15]: This change allows us to quickly find tests that failed.

Comment (harupy) [reply dev/run-python-flavor-tests.sh:15]: Let's say one of the tests in `tests/azureml` failed.

Before: we need to scroll back to the output log of `pytest tests/azureml`.
After: The output log looks like below:

```
...
============================= slowest 10 durations =============================
85.80s call     tests/h2o/test_h2o_model_export.py::test_pyfunc_serve_and_score
52.98s call     tests/shap/test_log.py::test_pyfunc_serve_and_score
45.46s call     tests/paddle/test_paddle_model_export.py::test_pyfunc_serve_and_score
12.01s call     tests/autologging/test_autologging_behaviors_integration.py::test_autolog_respects_silent_mode
11.58s call     tests/shap/test_log.py::test_log_model_with_extra_pip_requirements
10.74s call     tests/shap/test_log.py::test_sklearn_log_explainer
10.27s call     tests/shap/test_shap.py::test_log_explanation_with_classifier
9.02s setup    tests/shap/test_shap.py::test_log_explanation_with_classifier
8.40s call     tests/h2o/test_h2o_model_export.py::test_signature_and_examples_are_saved_correctly
8.09s call     tests/paddle/test_paddle_autolog.py::test_autolog_logs_expected_data
=========================== short test summary info ============================
FAILED tests/azureml/test_xxx.py::test_yyy
========== 1 failed, 1056 passed, 6189 warnings in 554.31s (0:09:14) ===========
```

Comment (WeichenXu123) [reply dev/run-python-flavor-tests.sh:15]: Good improvement!

File: tests/test_mlflow_lazily_imports_ml_packages.py [hunk 5]
```diff
@@ -1,23 +1,14 @@
...
+    "package",
+    (
         "catboost",
         "fastai",
-        "gluon",
+        "mxnet",
```
Comment (WeichenXu123) [inline tests/test_mlflow_lazily_imports_ml_packages.py:11]: why "gluon" removed ?

Comment (harupy) [reply tests/test_mlflow_lazily_imports_ml_packages.py:11]: See this line: https://github.com/mlflow/mlflow/blob/0bc7d2097270942d083c1168a1990c9091cb7ce4/mlflow/gluon/__init__.py#L76

Comment (harupy) [reply tests/test_mlflow_lazily_imports_ml_packages.py:11]: See this line: https://github.com/mlflow/mlflow/blob/0bc7d2097270942d083c1168a1990c9091cb7ce4/mlflow/gluon/__init__.py#L76

Comment (harupy) [reply tests/test_mlflow_lazily_imports_ml_packages.py:11]: `gluon` is a module in `mxnet`.

**Review by WeichenXu123 — APPROVED** [review]
LGTM

**Pull request merged**